src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,403 @@
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(startContext =>
+            {
+                var formatProviderType = startContext.Compilation.GetTypeByMetadataName(typeof(System.IFormatProvider).FullName);
+                if (formatProviderType == null)
+                {
+                    return;
+                }
+
+                startContext.RegisterSyntaxNodeAction(
+                    c => AnalyzeNode(c, formatProviderType),
+                    GetInvocationExpressionSyntaxKind());
+            });
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context, INamedTypeSymbol formatProviderType)
+        {
+            var optionSet = context.Options.GetDocumentOptionSetAsync(
+                context.Node.SyntaxTree, context.CancellationToken).GetAwaiter().GetResult();
+
+            if (optionSet.GetOption(
+                ValidateFormatStringOption.ReportInvalidPlaceholdersInStringDotFormatCalls,
+                context.SemanticModel.Language) == false)
+            {
+                return;
+            }
+
+            var syntaxFacts = GetSyntaxFactsService();
+            var expression = syntaxFacts.GetExpressionOfInvocationExpression(context.Node);
+
+            if (!IsValidFormatMethod(syntaxFacts, expression))
+            {
+                return;
+            }
+
+            var arguments = syntaxFacts.GetArgumentsOfInvocationExpression(context.Node);
+            var numberOfArguments = arguments.Count;
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(expression, context.CancellationToken);
+
+            var method = TryGetValidFormatMethodSymbol(context, numberOfArguments, symbolInfo);
+            if (method == null)
+            {
+                return;
+            }
+
+            var parameters = method.Parameters;
+
+            // If the chosen overload has a params object[] that is being passed an array of
+            // reference types then an actual array must be being passed instead of passing one
+            // argument at a time because we know the set of overloads and how overload resolution
+            // will work.  This array may have been created somewhere we can't analyze, so never
+            // squiggle in this case.
+            if (ArgsIsArrayOfReferenceTypes(context.SemanticModel, arguments, parameters, syntaxFacts))
+            {
+                return;
+            }
+
+            var formatStringLiteralExpressionSyntax = TryGetFormatStringLiteralExpressionSyntax(
+                context.SemanticModel,
+                arguments,
+                parameters,
+                syntaxFacts);
+
+            if (formatStringLiteralExpressionSyntax == null)
+            { 
+                return;
+            }
+
+            if (parameters.Length == 0)
```

Comment:
Shouldn't a string with a placeholder and no corresponding parameter be considered an error by this analyzer? Or am I misunderstanding this analyzer?